VehicleCode/Input/DataCollector.cpp

```diff
@@ -46,7 +55,65 @@ void DataCollector::checkSensors() {
 }
 
 
-// TEST define function
+/*!
+ * @brief checks the drive state and commands if the motor should be active or not
+ * 
+ * @param None
+ * @return None
+ */ 
+void DataCollector::checkDriveState() {
+
+    //INITIAL START
+    if(!driveState && brakeActive && switchActive && (brakeSensor != nullptr) && !startFault) {
+        driveState = !driveState;
+        sendLog(driveState);
+        brakeSensor->setDriveState();
+        Serial.println("INITIAL START");
+        return;
+    }
+
+    //FINAL STOP
+    if(driveState && !switchActive && (brakeSensor != nullptr)) {
+        driveState = !driveState;
+        sendLog(driveState);
+        brakeSensor->setDriveState();
+        brakeSensor->sendStopCommand();
+        Serial.println("FINAL STOP");
+        return;
+    }
+
+
+    //RUNNING STOP
+    if(!driveState && !switchActive && (brakeSensor != nullptr)) { // Allows for the timeout to not trip
+        brakeSensor->sendStopCommand();
+        return;
+    }
+}
+
+/*!
+ * @brief Sends the currentDrive State to the ECU (Car Class) Teensy so that it can 
+ * start logging
+ * 
+ * @param bool driveState (the current drive state being communicated)
+ * @return None
+ */ 
+
+void DataCollector::sendLog(bool driveState) {
```

Comment:
move msg definition to constructor as in old car class and only update elements as needed